sdk/examples/wallet/06_send_micro_transaction.rs

```diff
@@ -20,27 +20,31 @@ async fn main() -> Result<()> {
 
     // Get the account we generated with `01_create_wallet`
     let account = manager.get_account("Alice").await?;
-
-    // Set the stronghold password
-    manager
-        .set_stronghold_password(&env::var("STRONGHOLD_PASSWORD").unwrap())
-        .await?;
-
-    let outputs = vec![AddressWithMicroAmount {
-        address: "rms1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluaw60xu".to_string(),
-        amount: 1,
-        return_address: None,
-        expiration: None,
-    }];
-
-    let transaction = account.send_micro_transaction(outputs, None).await?;
-
-    println!(
-        "Transaction: {} Block sent: {}/api/core/v2/blocks/{}",
-        transaction.transaction_id,
-        &env::var("NODE_URL").unwrap(),
-        transaction.block_id.expect("no block created yet")
-    );
+    // May want to ensure the account is synced before sending a transaction.
+    let balance = account.sync(None).await?;
+
+    if balance.base_coin.available >= 1 {
```

Comment:
One could build an output like here https://github.com/iotaledger/iota-sdk/blob/aa68f6b6879ae17d661b53469e81add37d707592/sdk/examples/client/output/micro_transaction.rs#L55-L66, just with `BasicOutputBuilder::new_with_minimum_storage_deposit(rent_structure)?` (rent_structure can be retrieved with `client.get_rent_structure().await?`) and then check if the available amount is >= (even then you can't be sure, because if it's not ==, then you need a remainder output which also needs at least the minimum required storage deposit of an output)

So for example minimum required storage deposit is 1_000_000, your available balance is 2_000_000, then you want to send an output with 1_500_000, this would leave only 500_000 behind which would be not enough to cover the storage deposit

Reply:
Probably better to just try and fail then :/